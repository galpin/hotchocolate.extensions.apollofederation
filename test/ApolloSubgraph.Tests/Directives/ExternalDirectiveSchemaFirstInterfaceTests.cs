using System.Threading.Tasks;
using HotChocolate.Types;
using Microsoft.Extensions.DependencyInjection;
using Xunit;
using static HotChocolate.Extensions.ApolloSubgraph.Test;

namespace HotChocolate.Extensions.ApolloSubgraph.Directives;

public class ExternalDirectiveSchemaFirstInterfaceTests
{
    [Fact]
    public async Task When_external_is_specified_on_interface()
    {
        var schema = await BuildSchemaAsync(builder =>
        {
            builder.AddDocumentFromString(@"
                interface IProduct @key(fields: ""upc"") {
                     upc: String!
                     id: String @external
                }

                type Product @key(fields: ""upc"") {
                     upc: String!
                     id: String @external
                }

                type Query
            ");
        });

        var sut = schema.GetType<InterfaceType>("IProduct");

        Assert.Collection(
            sut.Fields["id"].Directives,
            x => Assert.Equal("external", x.Name));
        await schema.QuerySdlAndMatchSnapshotAsync();
    }
}