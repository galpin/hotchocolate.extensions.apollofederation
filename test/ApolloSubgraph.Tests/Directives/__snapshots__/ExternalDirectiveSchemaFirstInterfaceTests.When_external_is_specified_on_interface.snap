{
  "data": {
    "_service": {
      "sdl": "interface IProduct @key(fields: \u0022upc\u0022) {\n  upc: String!\n  id: String @external\n}\n\ntype Product @key(fields: \u0022upc\u0022) {\n  upc: String!\n  id: String @external\n}"
    }
  }
}
